Sources/PackageLoading/JSONSerialization.swift

```diff
@@ -8,115 +8,78 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
  */
 
-import Foundation
 import PackageDescription
+import Basic
 
 /// A JSON representation of an element.
 protocol JSONSerializable {
     
     /// Return a JSON representation.
-    func toJSON() -> AnyObject
+    func toJSON() -> JSON
 }
 
 public func jsonString(package: PackageDescription.Package) throws -> String {
-    let options: NSJSONWritingOptions = .prettyPrinted
 
-    let json: AnyObject = package.toJSON()
-    let data = try NSJSONSerialization.data(withJSONObject: json, options: options)
-    guard let string = String(data: data, encoding: NSUTF8StringEncoding) else { fatalError("NSJSONSerialization emitted invalid data") }
-    return string
-}
-
-extension NSMutableDictionary {
-    static func withNew(block: @noescape (dict: NSMutableDictionary) -> ()) -> NSMutableDictionary {
-        let dict = NSMutableDictionary()
-        block(dict: dict)
-        return dict
-    }
-    
-    func set(object: AnyObject, forKey key: String) {
-        self[key.asNS()] = object
-    }
-}
-
-extension String {
-    func asNS() -> NSString {
-        #if os(Linux)
-            return self.bridge()
-        #else
-            return self as NSString
-        #endif
-    }
-}
-
-extension Array {
-    func asNS() -> NSArray {
-        #if os(Linux)
-            return self.bridge()
-        #else
-            return self.map { $0 as! AnyObject } as NSArray
-        #endif
+    let json = package.toJSON()
+    guard let string = json.toBytes().asString else {
+        fatalError("Failed to serialize JSON \(json)")
```

Comment:
This shouldn't actually ever be possible (unless there was a bug in the JSON formatter in encoding bytes), so what I think we should get a `JSON.toString()` method that does it and doesn't fail.